### PR TITLE
gazebo_ros_planar_move fix - no more free-wheeling!

### PR DESCRIPTION
**Pull request by JacquesCloete**

### CODE REVIEW

**Review by wxmerkt — CHANGES_REQUESTED** [review]
Can you please include the provenance of the changes? E.g. if you started with a commit from a different repo / author, use cherry-pick and then make your amendments on top. Or if non-working, add note on who the original author was and where it's from and what has been modified in the longer git-commit message / comment

**Comment by JacquesCloete** [comment]
Ok, I added information on the origin of the fix in the commit. The original fix was implemented over a series of commits alongside other changes, so not sure how I'd cherry-pick

File: gazebo_plugins/src/gazebo_ros_planar_move.cpp [hunk 1]
```diff
@@ -139,7 +139,81 @@ namespace gazebo
...
+    rot_velocity_limit_min_ =-1.0;
+    x_acceleration_limit_max_ = 1.0;
+    x_acceleration_limit_min_ =-1.0;
+    y_acceleration_limit_max_ = 1.0;
+    y_acceleration_limit_min_ =-1.0;
+    rot_acceleration_limit_max_ = 1.0;
+    rot_acceleration_limit_min_ =-1.0;
+
+    if (sdf->HasElement("x_velocity_limit_max"))
+      (sdf->GetElement("x_velocity_limit_max")->GetValue()->Get(x_velocity_limit_max_));
```
Comment (wxmerkt) [inline gazebo_plugins/src/gazebo_ros_planar_move.cpp:175]: Why do you use these brackets around a statement?

File: gazebo_plugins/src/gazebo_ros_planar_move.cpp [hunk 2]
```diff
@@ -201,20 +285,72 @@ namespace gazebo
...
+    float dt  = (current_time-this->last_time_).Double();
+
+    ignition::math::Vector3d gravity(parent_->GetWorld()->Gravity());
+    ignition::math::Vector3d linear_vel = ignition::math::Matrix4d(pose.Rot()).Inverse() * parent_->WorldLinearVel();
+    ignition::math::Vector3d angular_vel = parent_->WorldAngularVel();
+
+    // check for joint_state_idel_sec_
+    if ( joint_state_idel_sec_ > 0 &&
+         (current_time-this->last_cmd_subscribe_time_).Double() > joint_state_idel_sec_ ) {
+      if (int((current_time-this->last_cmd_subscribe_time_).Double()*1000)%5000 < 1 ) {
```
Comment (wxmerkt) [inline gazebo_plugins/src/gazebo_ros_planar_move.cpp:297]: What do these magic numbers do? What for 5ms?

File: gazebo_plugins/src/gazebo_ros_planar_move.cpp [hunk 3]
```diff
@@ -201,20 +285,72 @@ namespace gazebo
...
+    this->GetNearestEntityBelow(fromEntity, distBelow, entityName);
+    if ( ( ! entityName.empty() ) && distBelow >  0.00001 ) {
+      linear_cmd_[2] += recover_z_velocity_p_gain_ * gravity[2] * dt;
+    }
+
+    // accel limit
+    ignition::math::Vector3d linear_acc  = (linear_cmd_  - last_linear_cmd_)/dt;
+    ignition::math::Vector3d angular_acc = (angular_cmd_ - last_angular_cmd_)/dt;
+
+    if ( linear_acc[0] > x_acceleration_limit_max_)  linear_cmd_[0] = last_linear_cmd_[0] + x_acceleration_limit_max_*dt;
```
Comment (wxmerkt) [inline gazebo_plugins/src/gazebo_ros_planar_move.cpp:321]: We might write these lines much more compact using an inline `clamp`

File: gazebo_plugins/src/gazebo_ros_planar_move.cpp (hunk 1 above)
Comment (JacquesCloete) [reply gazebo_plugins/src/gazebo_ros_planar_move.cpp:175]: Wasn't me, that was from the original fix. If they do nothing we can probably remove them.

File: gazebo_plugins/src/gazebo_ros_planar_move.cpp (hunk 2 above)
Comment (JacquesCloete) [reply gazebo_plugins/src/gazebo_ros_planar_move.cpp:297]: To be honest, I don't know - again, taken from k-okada's fix

File: gazebo_plugins/src/gazebo_ros_planar_move.cpp (hunk 3 above)
Comment (JacquesCloete) [reply gazebo_plugins/src/gazebo_ros_planar_move.cpp:321]: Ok cool

**Pull request merged**